packages/switch/test/lion-button-switch.test.js

```diff
@@ -0,0 +1,146 @@
+
+describe('lion-button-switch', () => {
+  let el;
+  beforeEach(async () => {
+    el = await fixture(html`
+      <lion-button-switch></lion-button-switch>
+    `);
+  });
+
+  it('should behave as a button', () => {
+    expect(el instanceof LionButton).to.be.true;
+  });
+
+  it('should have tabindex="0"', () => {
+    expect(el.tabIndex).to.equal(0);
+    expect(el.getAttribute('tabindex')).to.equal('0');
+  });
+
+  it('should have type="button"', () => {
+    expect(el.type).to.equal('button');
+    expect(el.getAttribute('type')).to.equal('button');
+  });
+
+  it('should have disabled=false by default', () => {
+    expect(el.disabled).to.equal(false);
+    expect(el.hasAttribute('disabled')).to.be.false;
+  });
+
+  it('should have checked=false by default', () => {
+    expect(el.checked).to.equal(false);
+    expect(el.hasAttribute('checked')).to.be.false;
+  });
+
+  it('should toggle the value of "checked" on click', async () => {
+    // given
+    expect(el.checked).to.be.false;
+    expect(el.hasAttribute('checked')).to.be.false;
+    // when
+    el.click();
+    await el.updateComplete;
+    // then
+    expect(el.checked).to.be.true;
+    expect(el.hasAttribute('checked')).to.be.true;
+    // when
+    el.click();
+    await el.updateComplete;
+    // then
+    expect(el.checked).to.be.false;
+    expect(el.hasAttribute('checked')).to.be.false;
+  });
+
+  it('should not toggle the valoue of "checked" on click if disabled', async () => {
+    // given
+    el.disabled = true;
+    expect(el.checked).to.be.false;
+    // when
+    el.click();
+    await el.updateComplete;
+    // then
+    expect(el.checked).to.be.false;
+    expect(el.hasAttribute('checked')).to.be.false;
+    // given
+    el.disabled = true;
+    el.checked = true;
+    // when
+    el.click();
+    await el.updateComplete;
+    // then
+    expect(el.checked).to.be.true;
+    expect(el.hasAttribute('checked')).to.be.true;
+  });
+
+  it('should dispatch "checked-changed" event', () => {
+    // given
+    const handlerSpy = sinon.spy();
+    el.addEventListener('checked-changed', handlerSpy);
+    // when
+    el.click();
+    el.click();
+    // then
+    expect(handlerSpy.callCount).to.equal(2);
+    const checkCall = (call, checked) => {
+      expect(call.args).to.have.a.lengthOf(1);
+      const e = call.args[0];
+      expect(e).to.be.an.instanceof(CustomEvent);
+      expect(e.bubbles).to.be.true;
+      expect(e.composed).to.be.true;
+      expect(e.detail).to.equal(checked);
+    };
+    checkCall(handlerSpy.getCall(0), true);
+    checkCall(handlerSpy.getCall(1), false);
+  });
+
+  it('should not dispatch "checked-changed" event if disabled', () => {
+    // given
+    const handlerSpy = sinon.spy();
+    el.disabled = true;
+    el.addEventListener('checked-changed', handlerSpy);
+    // when
+    el.click();
+    // then
+    expect(handlerSpy.called).to.be.false;
+  });
+
+  describe('a11y', () => {
+    describe('aria-pressed', () => {
```

Comment:
It's pretty verbose. I like that it's thorough though. I wonder if it would be nicer in a single it
```js
// expect that checks the default
el.click()
// expect that checks attr
el.checked = false;
// expect that checks attr
el.setAttribute('checked', true);
// expect that checks attr
```